src/Stan/Hie/MatchAst.hs

```diff
@@ -54,37 +54,37 @@ hieMatchPatternAst hie@HieFile{..} node@Node{..} = \case
            hieMatchPatternAst hie node p1
         && hieMatchPatternAst hie node p2
     PatternAstConstant lit ->
-           Set.member literalAnns (Set.map toNodeAnnotation (nodeAnnotations nodeInfo))
-        && ( let span = slice nodeSpan hie_hs_src in case lit of
-                ExactNum n   -> (span >>= readMaybe . decodeUtf8) == Just n
-                ExactStr s   -> span == Just s
-                PrefixStr s  -> maybe False (s `BS.isPrefixOf`) span
-                ContainStr s -> maybe False (s `BS.isInfixOf`) span
+           Set.member literalAnns (Set.map toNodeAnnotation (nodeAnnotations nodeInfo'))
+        && ( let span' = slice nodeSpan hie_hs_src in case lit of
+                ExactNum n   -> (span' >>= readMaybe . decodeUtf8) == Just n
+                ExactStr s   -> span' == Just s
+                PrefixStr s  -> maybe False (s `BS.isPrefixOf`) span'
+                ContainStr s -> maybe False (s `BS.isInfixOf`) span'
                 AnyLiteral   -> True
            )
     PatternAstName nameMeta patType ->
         any (matchNameAndType nameMeta patType)
         $ Map.assocs
-        $ nodeIdentifiers nodeInfo
+        $ nodeIdentifiers nodeInfo'
     PatternAstNode tags ->
-        matchAnnotations tags nodeInfo
+        matchAnnotations tags nodeInfo'
     PatternAstNodeExact tags patChildren ->
-           matchAnnotations tags nodeInfo
+           matchAnnotations tags nodeInfo'
         && checkWith (hieMatchPatternAst hie) nodeChildren patChildren
     PatternAstVarName varName -> isJust $ find
         (\case
             Right x -> varName `Str.isInfixOf` map toLower (occNameString $ nameOccName x)
             Left _ -> False
         )
-        $ Map.keys $ nodeIdentifiers nodeInfo
+        $ Map.keys $ nodeIdentifiers nodeInfo'
     PatternAstIdentifierDetailsDecl declType -> any (any (isDecl declType) . identInfo) $
-        Map.elems $ nodeIdentifiers nodeInfo
+        Map.elems $ nodeIdentifiers nodeInfo'
   where
     matchAnnotations :: Set NodeAnnotation -> NodeInfo TypeIndex -> Bool
     matchAnnotations tags NodeInfo{..} =
       tags `Set.isSubsetOf` Set.map toNodeAnnotation nodeAnnotations
 
-    nodeInfo = Stan.Hie.Compat.nodeInfo node
+    nodeInfo' = Stan.Hie.Compat.nodeInfo node
```

Comment:
Would you mind undoing this one? Or alternatively add `import qualified Stan.Hie.Compat`.

Reply:
@tomjaguarpaw, that is not the source of the shadowing problem that the `'` fixes. The problem is that for `ghc-8.10.7` (and maybe other versions) `nodeInfo` is a field of the data constructor of type `HieTypes.HieAST a`.

Reply:
Ah I see. Let's keep this one then.